runners/flink/src/test/java/org/apache/beam/runners/flink/FlinkStreamingPipelineTranslatorTest.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.flink;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertNull;
+
+import java.util.List;
+import java.util.UUID;
+import org.apache.beam.runners.flink.FlinkStreamingPipelineTranslator.FlinkAutoBalancedShardKeyShardingFunction;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.util.SerializableUtils;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.ShardedKey;
+import org.apache.beam.vendor.guava.v20_0.com.google.common.collect.Lists;
+import org.junit.Test;
+
+/** Tests if overrides are properly applied. */
+public class FlinkStreamingPipelineTranslatorTest {
+
+  @Test
+  public void testAutoBalanceShardKeyResolvesMaxParallelism() {
+
+    assertThat(
+        new FlinkAutoBalancedShardKeyShardingFunction<>(3, -1, StringUtf8Coder.of())
+            .getMaxParallelism(),
+        equalTo(128));
```

Comment:
`equalTo(FlinkKeyGroupAssignment.getMaxParallelism())`?

Reply:
I can not find `` ... Can only do `equalTo(KeyGroupRangeAssignment.computeDefaultMaxParallelism(parallelism))` what is done in sharding function, if you like it better than hardcoded value.

Reply:
Yes, that's what I meant.